docs/cpp/cpp-ide.md

```diff
@@ -49,74 +51,105 @@ ColumnLimit: 0
+
+![Peek definition](images/cpp/peekdefn.png)
+
+With the peek window open, you browse the list of results shown to find the one you're interested in. If you want to navigate to the location of one of these results, just select result or double-click in the source code displayed on the left-hand side of the peek window.
+
+### Go to definition
+
+Use the **Go to Definition** feature to quickly navigate to where a symbol is defined in your source code. Select a symbol in your source code and then press `kb(editor.action.revealDefinition)`, or right-click and choose **Go to Definition** from the context menu. When there's only one definition of the symbol, you'll navigate directly to its location, otherwise the competing definitions are displayed in a peek window as described in the previous section.
+
+If no definitions can be found for the symbol you selected, the C/C++ Extension will automatically search for a declaration of the symbol.
+
+### Go to declaration
+
+Use the **Go to Declaration** feature to navigate to the location a symbol is declared in your source code. This feature functions the same as **Go to Definition**, but for declarations. Select a symbol in your source code, right-click, and choose **Go to Declaration** from the context menu. This will navigate you to the location of the symbol's declaration.
+
+### Go to references
 
-### Go to Definition
+Use the **Go to References** feature to understand how often and where a symbol is referenced in your source code. Select a symbol in your source code and press `kb(editor.action.goToReferences)` or right-click and choose **Go to References** from the context menu. If any references are found, they are displayed in a peek window.
 
-You can also quickly navigate to where a symbol is defined by using the **Go to Definition** feature.
+### Go to type definition
 
-To go to a symbol's definition, place your cursor on the symbol anywhere it is used in your source code and then press `kb(editor.action.revealDefinition)`. Or, choose **Go to Definition** from the context menu (right-click, then choose **Go to Definition**). When there's only one definition of the symbol, you'll navigate directly to its location, otherwise the competing definitions are displayed in a peek window as described in the previous section and you have to choose the definition that you want to go to.
+Use the **Go to Type Definition** feature to jump to where a type is defined in your source code. Select a type in your source code and press `kb(editor.action.goToTypeDefinition)` or right-click and choose **Go to Type Definition** from the context menu.
 
 ## Next steps
 
 Read on to find out about:
 
-* [Configure VS Code for Windows Subsystem for Linux](/docs/cpp/config-wsl.md)
+* [Debugging](/docs/cpp/cpp-debug.md) - find out how to use the debugger with your project
```

Comment:
Could you unify the style for the text in this list? Some bullet points start with a capital letter other don't and others end with a period and others don't. Maybe they should all be sentence styled?